src/main/scala/se/nimsa/sbx/app/routing/ImageRoutes.scala

```diff
@@ -59,58 +60,46 @@ trait ImageRoutes {
           }
         }
       } ~ pathPrefix(LongNumber) { imageId =>
-        onSuccess(metaDataService.ask(GetImage(imageId)).mapTo[Option[Image]]) {
-          case Some(image) =>
-            pathEndOrSingleSlash {
-              get {
-                complete(HttpEntity(`application/octet-stream`, storage.fileSource(image)))
-              } ~ delete {
-                complete(metaDataService.ask(DeleteMetaData(image)).map { _ =>
-                  system.eventStream.publish(ImageDeleted(image.id))
-                  storage.deleteFromStorage(image)
-                  NoContent
-                })
-              }
-            } ~ path("attributes") {
-              get {
-                complete(storage.readImageAttributes(image))
-              }
-            } ~ path("imageinformation") {
-              get {
-                complete(storage.readImageInformation(image))
-              }
-            } ~ path("png") {
-              parameters((
-                'framenumber.as[Int] ? 1,
-                'windowmin.as[Int] ? 0,
-                'windowmax.as[Int] ? 0,
-                'imageheight.as[Int] ? 0)) { (frameNumber, min, max, height) =>
-                get {
-                  onComplete(Future(storage.readPngImageData(image, frameNumber, min, max, height))) {
-                    case Success(bytes) => complete(HttpEntity(`image/png`, bytes))
-                    case Failure(_) => complete(NoContent)
-                    case _ => complete(InternalServerError)
-                  }
-                }
+        pathEndOrSingleSlash {
+          get {
+            complete(HttpEntity(`application/octet-stream`, storage.fileSource(imageId)))
+          } ~ delete {
+            complete(metaDataService.ask(DeleteMetaData(Seq(imageId))).map { _ =>
+              system.eventStream.publish(ImagesDeleted(Seq(imageId)))
+              storage.deleteFromStorage(Seq(imageId))
+              NoContent
+            })
+          }
+        } ~ path("attributes") {
+          get {
+            complete(storage.readImageAttributes(imageId))
+          }
+        } ~ path("imageinformation") {
+          get {
+            complete(storage.readImageInformation(imageId))
+          }
+        } ~ path("png") {
+          parameters((
+            'framenumber.as[Int] ? 1,
+            'windowmin.as[Int] ? 0,
+            'windowmax.as[Int] ? 0,
+            'imageheight.as[Int] ? 0)) { (frameNumber, min, max, height) =>
+            get {
+              onComplete(Future(storage.readPngImageData(imageId, frameNumber, min, max, height))) {
+                case Success(bytes) => complete(HttpEntity(`image/png`, bytes))
+                case Failure(e) if e.isInstanceOf[NotFoundException] => complete(NotFound)
+                case Failure(_)  => complete(NoContent)
```

Comment:
Thanks for pointing this out. Although we are picky with the SOP Classes and transfer syntaxes we support, some datasets cannot be rendered as PNG, and users can always allow more SOP Classes. But returning NoContent in those cases is too benign. I changed this to 501 NotImplemented and updated the API spec.